x/cdp/keeper/fees.go

```diff
@@ -27,6 +27,38 @@ func (k Keeper) CalculateFees(ctx sdk.Context, principal sdk.Coins, periods sdk.
 	return newFees
 }
 
+// UpdateFeesForRiskyCdps calculates fees for risky CDPs
+// The overall logic is first select the CDPs with 10% of the liquidation ratio
+// Then we call calculate fees on each of those CDPs
+// Next we store the result of the fees in the cdp.AccumulatedFees field
+// Finally we set the cdp.FeesUpdated time to the current block time (ctx.BlockTime()) since that
+// is when we made the update
+func (k Keeper) UpdateFeesForRiskyCdps(ctx sdk.Context, collateralDenom string) sdk.Error {
```

Comment:
```suggestion
func (k Keeper) UpdateFeesForRiskyCdps(ctx sdk.Context, collateralDenom string, marketID string) sdk.Error {
        price, err := k.pricefeedKeeper.GetCurrentPrice(ctx, marketID)
	if err != nil {
		return err
	}
	// first calculate the target ratio based on liquidation ratio plus ten percent
	value, err := sdk.NewDecFromStr("1.1")
	if err != nil {
		return err
	}
	targetRatio := k.getLiquidationRatio(ctx, collateralDenom).Mul(value) // corresponds to 110% of the liquidation ratio
        normalizedRatio := sdk.OneDec().Quo(price.Price.Quo(liquidationRatio))
        
	// now iterate over all the cdps based on collateral ratio
	k.IterateCdpsByCollateralRatio(ctx, collateralDenom, normalizedRatio, func(cdp types.CDP) bool {
```

CDPs need to be selected based on the collateral:debt ratio. Currently, we are selecting based on the usd_value_of_collateral:debt ratio. This suggestions gives the correct approach. Note that `CollateralParam` contains `MarketID` as a field, so we can call the function with `cp.MarketID`.